app/models/enrollment.rb

```diff
@@ -28,6 +28,90 @@ def contain_this_time_slot?(time_slot)
     start_time <= time_slot && time_slot < end_time
   end
 
+  def check_if_member_is_already_enrolled
+    if mission.members.include?(member)
+      failure_message = I18n.t('activerecord.errors.models.enrollment.member_already_enrolled')
+      errors.add :member_already_enrolled, failure_message
+      return false
+    end
+
+    true
+  end
+
+  def check_if_the_standard_mission_is_not_full
+    return true unless mission.genre == 'standard'
+
+    if mission.standard_full?
+      failure_message = I18n.t('activerecord.errors.models.enrollment.full_mission')
+      errors.add :full_mission, failure_message
+      return false
+    end
+
+    true
+  end
+
+  def check_if_the_duration_is_positive
+    if start_time >= end_time
+      failure_message = I18n.t('activerecord.errors.models.enrollment.negative_duration')
+      errors.add :negative_duration, failure_message
+      return false
+    end
+
+    true
+  end
+
+  def duration_multiple_of_90_minutes?
+    ((end_time.to_i - start_time.to_i) % (60 * 90)).zero?
+  end
+
+  def check_if_datetimes_of_enrollment_are_inside_the_mission_s_period
+    unless mission.inside_period?(self)
+      failure_message = I18n.t('activerecord.errors.models.enrollment.inconsistent_datetimes')
+      errors.add :inconsistent_datetimes, failure_message
+      return false
+    end
```

Comment:
Since an Enrollment belongs to a Mission, and you are asking here to the Mission if the current enrollment is valid; the responsiblity should be kept in the Enrollment.
You could refactor this as
```ruby
  def check_if_datetimes_of_enrollment_are_inside_the_mission_s_period
    unless inside_mission_period?
      failure_message = I18n.t('activerecord.errors.models.enrollment.inconsistent_datetimes')
      errors.add :inconsistent_datetimes, failure_message
      return false
    end
  end

  private

  def inside_mission_period?
    start_time >= mission.start_date &&
      start_time <= mission.due_date &&
      end_time >= mission.start_date &&
      end_time <= mission.due_date
  end